sql-plugin/src/main/scala/com/nvidia/spark/rapids/profiler.scala

```
@@ -0,0 +1,387 @@
+import java.util.concurrent.{ConcurrentHashMap, Future, ScheduledExecutorService, TimeUnit}
+import java.util.regex.Pattern
+
+import scala.collection.mutable
+
+import com.nvidia.spark.rapids.jni.Profiler
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.ipc.CallerContext
+
+import org.apache.spark.{SparkContext, TaskContext}
+import org.apache.spark.api.plugin.PluginContext
+import org.apache.spark.internal.Logging
+import org.apache.spark.io.CompressionCodec
+import org.apache.spark.scheduler.{SparkListener, SparkListenerJobEnd, SparkListenerStageCompleted}
+import org.apache.spark.sql.rapids.execution.TrampolineUtil
+import org.apache.spark.util.SerializableConfiguration
+
+object ProfilerOnExecutor extends Logging {
+  private val jobPattern = Pattern.compile("SPARK_.*_JId_([0-9]+).*")
+  private var writer: Option[ProfileWriter] = None
+  private var timeRanges: Option[Seq[(Long, Long)]] = None
+  private var jobRanges: RangeConfMatcher = null
+  private var stageRanges: RangeConfMatcher = null
+  // NOTE: Active sets are updated asynchronously, synchronize on ProfilerOnExecutor to access
+  private val activeJobs = mutable.HashSet[Int]()
+  private val activeStages = mutable.HashSet[Int]()
+  private var timer: Option[ScheduledExecutorService] = None
+  private var timerFuture: Option[Future[_]] = None
+  private var driverPollMillis = 0
+  private val startTimestamp = System.nanoTime()
+  private var isProfileActive = false
+
+  def init(pluginCtx: PluginContext, conf: RapidsConf): Unit = {
+    require(writer.isEmpty, "Already initialized")
+    timeRanges = conf.profileTimeRangesSeconds.map(parseTimeRanges)
+    jobRanges = new RangeConfMatcher(conf, RapidsConf.PROFILE_JOBS)
+    stageRanges = new RangeConfMatcher(conf, RapidsConf.PROFILE_STAGES)
+    driverPollMillis = conf.profileDriverPollMillis
+    if (timeRanges.isDefined && (stageRanges.nonEmpty || jobRanges.nonEmpty)) {
+      throw new UnsupportedOperationException(
```

Does the job crash if this happens?  Would it be better to just have a priority on which one to use, and then WARN the user if it happens/possibly add a warning to the file being written? Especially when other places will eat exceptions from the timer and just log them. But I understand why that might be happening, so I can see it either way.